central/role/datastore/singleton.go

```diff
@@ -67,23 +67,37 @@ func Singleton() DataStore {
 
 type roleAttributes struct {
 	idSuffix           string
+	postgresID         string
 	description        string
 	resourceWithAccess []permissions.ResourceWithAccess
 }
 
+func (attributes *roleAttributes) getID() string {
+	if env.PostgresDatastoreEnabled.BooleanSetting() {
+		return attributes.postgresID
+	}
+	return rolePkg.EnsureValidPermissionSetID(attributes.idSuffix)
+}
+
+// The postgresID field is there for UUID support of permission set ID fields in postgres mode.
+// The values are UUIDs taken in descending order from ffffffff-ffff-fff4-f5ff-ffffffffffff
+// Next ID: ffffffff-ffff-fff4-f5ff-fffffffffff6
 var defaultRoles = map[string]roleAttributes{
 	rolePkg.Admin: {
 		idSuffix:           "admin",
+		postgresID:         "ffffffff-ffff-fff4-f5ff-ffffffffffff",
```

Comment:
If I read the spec correctly, there are only 6 bits that distinguish v4 UUIDs from random 128-bit strings, and they should be set accordingly here.
Now, v4 is supposedly for random UUIDs, while the IDs used for the default permissions and scopes are fixed.

IMHO, the IDs for the default permission sets and access scopes should have fixed values so that the role references to these do not get broken.